homeassistant/components/cmus/media_player.py

```diff
@@ -80,18 +80,41 @@ def __init__(self, server, password, port, name):
         """Initialize the CMUS device."""
 
         if server:
-            self.cmus = remote.PyCmus(server=server, password=password, port=port)
             auto_name = f"cmus-{server}"
         else:
-            self.cmus = remote.PyCmus()
             auto_name = "cmus-local"
         self._name = name or auto_name
+        self._server = server
+        self._password = password
+        self._port = port
         self.status = {}
+        self.connect()
+
+    def connect(self):
+        """Connect the CMUS device."""
+
+        try:
+            if self._server:
+                self.cmus = remote.PyCmus(
+                    server=self._server, password=self._password, port=self._port
+                )
+            else:
+                self.cmus = remote.PyCmus()
+        except exceptions.InvalidPassword:
+            _LOGGER.error("The provided password was rejected by cmus")
+        except exceptions.CmusNotRunning:
+            _LOGGER.error("cmus does not appear to be running")
 
     def update(self):
         """Get the latest data and update the state."""
-        status = self.cmus.get_status_dict()
-        if not status:
+        try:
+            status = self.cmus.get_status_dict()
+        except BrokenPipeError:
+            self.connect()
+        except exceptions.ConfigurationError:
+            self.connect()
+            _LOGGER.warning("A configuration error occurred")
+        if "status" not in locals():
```

Comment:
Safely check the status is set (in case it's not defined here). `if not status` was throwing exceptions here.